### DISCUSSION TITLE
how to render this with the other framework?

### DISCUSSION THREAD

**Discussion by Albert-Gao** [react-grid-layout/react-grid-layout#1558] [Q&A]
Thanks for this lib! If I want to render the layout in plain html, how to do it? Thanks :)

**Comment by STRML** [comment]
I'm sorry but I don't understand this question. Are you asking how to render React to HTML? If so, see https://reactjs.org/docs/react-dom-server.html

**Reply by Albert-Gao** [reply]
My bad, so here is my requirement,

1. made a fluid grid (done by this framework, works flawlessly)
2. serialize it into JSON (I save all the `data-grid` object)
3. render this JSON with another framework

I think, for step 3, if i can create a DEMO where the JSON can be display as UI with plain html/js/css, that should be enough.

To make it more clear, if I have `layout` info for all children in this image:
![image](https://user-images.githubusercontent.com/18282328/132787961-1e4f4572-619c-4e11-8732-b77031da77b3.png)

I saved all layout information and element attributes into a json like 

```javascript
const componentsJson = {
  components: {
    tag: 'div',
    attributes: {
      border: '1px solid black',
    },
    layout: {
      w: 4,
      h: 13,
      x: 0,
      y: 0,
      i: 'chakara:Stack:Column:01FETQ5ESR8QJ0DA4A0MHB4DWR',
      moved: false,
      static: false,
      isDraggable: true,
      isResizable: true,
      isBounded: true,
    },
    children: {
      // which just another component
    },
  },
}
```

Is there a way that I can recreate the UI in a plain HTML/CSS way?

My requirement is to persist the layout as a json and render it in another framework :)

The reason I ask this, is seems saving the layout object is not enough, there are styles generated during runtime by `react-grid-layout`

![image](https://user-images.githubusercontent.com/18282328/132788348-40de0809-6fe0-4e9d-8e53-bfe5e6829e9e.png)

I was thinking like wrapping the children with `<div class="react-grid-layout"></div>` and recreate the children elements with a `<div class="react-grid-item react-draggable cssTransforms react-resizable">` wrapper, but seems I need to save the inline-styles as well?

Is there a way I can persist the whole layout and reused later (without the requirement of drag-and-drop)?
